sdk/storage/storage-file-share/test/fileclient.spec.ts

```diff
@@ -2944,3 +2948,438 @@ describe("FileClient - AllowTrailingDots - Default", () => {
+    assert.deepEqual(result.fileMode, posixProperties.fileMode);
+    assert.deepEqual(result.linkCount, 1);
+  });
+
+  it("startCopy - with NFS properties", async function () {
+    const posixProperties: FilePosixProperties = {
+      owner: "123",
+      group: "654",
+      fileMode: {
+        owner: {
+          read: true,
+          write: true,
+          execute: true,
+        },
+        group: {
+          read: true,
+          write: false,
+          execute: true,
+        },
+        other: {
+          read: true,
+          write: false,
+          execute: true,
+        },
+        effectiveUserIdentity: false,
+        effectiveGroupIdentity: false,
+        stickyBit: false,
+      },
+      fileType: "Regular",
+    };
+
+    await fileClient.create(1024);
+    const newFileClient = dirClient.getFileClient(
+      recorder.variable("copiedfile", getUniqueName("copiedfile")),
+    );
+
+    const result = await newFileClient.startCopyFromURL(fileClient.url, {
+      posixProperties: posixProperties,
+      fileModeCopyMode: "override",
+      fileOwnerCopyMode: "override",
+    });
+    assert.ok(result.copyId);
+
+    const properties = await newFileClient.getProperties();
+    assert.deepEqual(properties.owner, posixProperties.owner);
+    assert.deepEqual(properties.group, posixProperties.group);
+    assert.deepEqual(properties.fileMode, posixProperties.fileMode);
+    assert.deepEqual(properties.linkCount, 1);
+  });
+
+  it("startCopy - with default", async function () {
+    const posixProperties: FilePosixProperties = {
+      owner: "123",
+      group: "654",
+      fileMode: {
+        owner: {
+          read: true,
+          write: true,
+          execute: true,
+        },
+        group: {
+          read: true,
+          write: false,
+          execute: true,
+        },
+        other: {
+          read: true,
+          write: false,
+          execute: true,
+        },
+        effectiveUserIdentity: false,
+        effectiveGroupIdentity: false,
+        stickyBit: false,
+      },
+      fileType: "Regular",
+    };
+    await fileClient.create(1024, { posixProperties: posixProperties });
+
+    const getResp = await fileClient.getProperties();
+    assert.deepEqual(getResp.owner, posixProperties.owner);
+    assert.deepEqual(getResp.group, posixProperties.group);
+    assert.deepEqual(getResp.fileMode, posixProperties.fileMode);
+    assert.deepEqual(getResp.nfsFileType, posixProperties.fileType);
+    assert.deepEqual(getResp.linkCount, 1);
+
+    const newFileClient = dirClient.getFileClient(
+      recorder.variable("copiedfile", getUniqueName("copiedfile")),
+    );
+    const result = await newFileClient.startCopyFromURL(fileClient.url);
+    assert.ok(result.copyId);
+
+    const properties = await newFileClient.getProperties();
+    assert.deepEqual(properties.owner, "0");
+    assert.deepEqual(properties.group, "0");
+    assert.ok(properties.fileMode);
```

Comment:
should check the filemode is default value?

Reply:
Updated